code/client/src/util.js

```diff
@@ -71,6 +72,32 @@ export default {
     const fiat = (price || 0) * parseFloat(ones)
     const fiatFormatted = exports.default.formatNumber(fiat)
     return { balance, formatted, fiat, fiatFormatted, valid: true }
+  },
+
+  validateAddress: (address) => {
```

Comment:
I see error message is prompted in this function instead of at each component themselves. And this function actually convert the address to ETH form, not just validate an address. Perhaps it should be renamed to `normalizeAddress` or something similar, and the UI prompt should be moved out to another function (e.g. `promptBadAddress`)